app/models/cloud_volume.rb

```diff
@@ -99,6 +99,7 @@ def self.create_volume(ems_id, options = {})
   def self.validate_create_volume(ext_management_system)
     klass = class_by_ems(ext_management_system)
     return klass.validate_create_volume(ext_management_system) if ext_management_system &&
+                                                                  klass != self &&
```

Comment:
Should this be using supports feature instead?  I don't really understand this validate_ code.

Reply:
Being converted to use SupportsFeatureMixin in PR Set starting with https://github.com/ManageIQ/manageiq/pull/21393

Reply:
the problem is if this does not have a type but does have an ems, then it will end up in an infinite loop
Maybe this is just a test issue, but it is nailing us in manageiq-api

Reply:
Yeah I'd prefer to merge https://github.com/ManageIQ/manageiq/pull/21393, let me see if I hit the same issue as @kavyanekkalapu on the UI PR, otherwise I'll just merge it

Reply:
also, for the record: we do not use this validation pattern anywhere

Think this is wrong.

It is the same code in the api and in ui-classic

Reply:
Looks like @chessbyte's patch set has fixed this, https://app.travis-ci.com/github/ManageIQ/manageiq-api/jobs/539531336

Reply:
> also, for the record: we do not use this validation pattern anywhere
>
> Think this is wrong.

This code is bad, and you should feel bad :wink: